src/olympia/files/tasks.py

```diff
@@ -35,14 +35,17 @@ def extract_webext_permissions(ids, **kw):
                      (file_.pk, file_.current_file_path))
             parsed_data = parse_xpi(file_.current_file_path, user=user)
             permissions = parsed_data.get('permissions', [])
+            optional_permissions = parsed_data.get('optional_permissions', [])
             # Add content_scripts host matches too.
             for script in parsed_data.get('content_scripts', []):
                 permissions.extend(script.get('matches', []))
-            if permissions:
+            if permissions or optional_permissions:
                 log.info('Found %s permissions for: %s' %
                          (len(permissions), file_.pk))
```

Comment:
We should probably update this log message to include optional permissions.

Reply:
Taking another look at this, I have a feeling that the `extract_permissions` command, and this related `extract_webext_permissions` are obsolete, so rather than update the code and tests to address optional permissions, perhaps we should just delete the command and task. WDYT @diox?

Reply:
ha yeah, it seems to be called in a management command but maybe we don't need it anymore

Reply:
We discussed this on Matrix, and we do in fact need this to extract the optional permissions to backfill all the add-ons, but some additional changes will be needed. I'm going to defer that to a follow-up issue.

Reply:
I opened https://github.com/mozilla/addons-server/issues/15062. All outstanding issues for this PR have been addressed. I'll merge it once Travis is happy.